libkbfs/file_data.go

```diff
@@ -509,44 +535,298 @@ func (fd *fileData) createIndirectBlock(
 	return fblock, nil
 }
 
-// newRightBlock appends a new block (with a temporary ID) to the end
-// of this file.
+// newRightBlock creates space for a new rightmost block, creating
```

Comment:
I don't think it claims anywhere to append to the end of the file, and this PR doesn't change that.  It's just making a rightmost leaf block under the given offset, and it's the callers responsibility to move it into the correct place in the tree.  I'll try to clarify the comment.